src/nvim/os/env.c

```diff
@@ -87,3 +97,598 @@ void os_get_hostname(char *hostname, size_t len)
 #endif
 }
 
+/// To get the "real" home directory:
+///   - get value of $HOME
+/// For Unix:
+///   - go to that directory
+///   - do os_dirname() to get the real name of that directory.
+/// This also works with mounts and links.
+/// Don't do this for MS-DOS, it will change the "current dir" for a drive.
```

Comment:
Hmmm... the answer seems to be "sometimes, except when it isn't," so I think the best solution is to leave it up to somebody who actually knows the Win32 API to decide.